### PR TITLE
Tsmith/update functional test

### PR DESCRIPTION
**Pull request by Bento007**
#### Reviewers
**Functional:** @MDunitz 

**Readability:** @tihuan 

---

## Changes
- add functional test for updating a collection
- add functional test cannot delete a public collection.

### CODE REVIEW

File: tests/functional/backend/corpora/test_api.py [hunk 1]
```diff
@@ -183,4 +233,4 @@ def test_dataset_upload_flow(self):
 
         # Check that the dataset is gone
         res = requests.get(f"{self.api}/dp/v1/datasets/{dataset_uuid}/status", headers=headers)
-        self.assertEqual(res.status_code, requests.codes.unauthorized)
+        self.assertEqual(res.status_code, requests.codes.forbidden)
```
Comment (tihuan) [inline tests/functional/backend/corpora/test_api.py:236]: nit: should this be 404 not found instead of forbidden, if the dataset is gone?

**Review by tihuan — APPROVED** [review]
All LGTM! Just one tiny question 😸 Thank you!

File: tests/functional/backend/corpora/test_api.py (hunk 1 above)
Comment (Bento007) [reply tests/functional/backend/corpora/test_api.py:236]: This is the correct value for now. We will probably change it to something else once we return "removed by user" for deleted public collections.

Comment (tihuan) [reply tests/functional/backend/corpora/test_api.py:236]: Ahh that's right! Thanks Trent 😄

**Review by tihuan — APPROVED** [review]
LGTM!!

**Review by MDunitz — APPROVED** [review]
lgtm

**Pull request merged**